### PR TITLE
Write tests for boolean group broken

### PR DESCRIPTION
**Pull request by binarygit**
# Description

This PR adds tests for the broken boolean group fix PR. 

It also makes sure such boolean group's check boxes are correctly checked in the edit view.

annotely_image

# Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Manual review steps

1. Step 1
1. Step 2

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

File: spec/dummy/app/models/user.rb [hunk 1]
```diff
@@ -97,8 +99,4 @@ def permissions
       delete: true,
     }
   end
-
-  def permissions=(value)
-    # assert here the value is a hash and has the right values sent in by the user
-  end
```
Comment (binarygit) [inline spec/dummy/app/models/user.rb:103]: @adrianthedev I thought it'd be nicer to not have a body for this at all and just have this method created via the setter. WDYT?

**Comment by Paul-Bob** [comment]
@binarygit, Let's keep the old tests for roles to ensure nothing breaks. Let's instead add some new test cases for permissions without modifying the roles tests.

**Comment by binarygit** [comment]
All the old tests are there. I just refactored them a little bit and added the new tests but the diff is showing that I changed the whole thing for some reason.

File: spec/dummy/app/models/user.rb (hunk 1 above)
Comment (adrianthedev) [reply spec/dummy/app/models/user.rb:103]: Oh, so basically set the value on the model instance class. 
Yeah, that could work too. 
Please as a comment on the writer why it's there.

**Review by adrianthedev — APPROVED** [review]
Looks good, thank you!

**Pull request merged**